manuskript/ui/editors/editorWidget.py

```diff
@@ -344,12 +352,33 @@ def modelDataChanged(self, topLeft, bottomRight):
         if topLeft.row() <= self.currentIndex.row() <= bottomRight.row():
             self.updateStatusBar()
 
-    #def rowsAboutToBeRemoved(self, parent, first, last):
-        #if self.currentIndex:
-            #if self.currentIndex.parent() == parent and \
-                                    #first <= self.currentIndex.row() <= last:
-                ## Item deleted, close tab
-                #self.mw.mainEditor.tab.removeTab(self.mw.mainEditor.tab.indexOf(self))
+    def rowsInserted(self, parent, first, last):
+        if self.currentIndex:
+            self.updateIndexFromID(fallback=self.currentIndex.parent())
+
+    def rowsAboutToBeRemoved(self, parent, first, last):
+        if not self.currentIndex:
+            return
+
+        # Check parents until we find a parent that matches the parent we are removing.
```

Comment:
Comment is confusing, maybe rewrite into : 
`# Look for a common parent to verify if the deleted rows include our index in their hierarchy`

Reply:
I have rewritten it into '_Look for a common ancestor to verify whether the deleted rows include our index in their hierarchy._' to avoid the entire 'parent' word entirely.

Reply:
Good idea!